### DISCUSSION TITLE
Adding curl to Dockerfiles

### DISCUSSION THREAD

**Discussion by rrauenza** [sorry-cypress/sorry-cypress#645] [Ideas]
PR:  https://github.com/sorry-cypress/sorry-cypress/pull/643

I was writing HEALTHCHECKs for my docker compose, but found that I was missing curl.  To add curl in a derived Dockerfile I need to switch USER to root, run apk, and then switch back to USER node.  

...but this assumes that the parent image uses node as $USER, which could change in the future.  

..Alternatively, you could put USER=node in the ENV and we could use that to switch back to the canonical user in the derived images.

Thoughts?

**Comment by agoldis** [comment]
Sorry for the delayed response. Not sure what are the implications, I remember people wanted it to be root-less for compatibility with OpenShift.

**Reply by rrauenza** [reply]
Maybe then we could just set ARG USER in the Dockerfile then so that it can be referenced later to go back to it...

**Comment by rrauenza** [comment]
To be clear, because it wasn't to me until I refreshed my memory.. the request is either

(1) add curl to docker images for HEALTHCHECK
or
(2) add ARG USER=node to allow us to switch back to proper user in derived images.

These would be in api/Dockerfile dashboard/Dockerfile and director/Dockerfile

I can generate a PR based on the direction you'd prefer...

**Reply by rrauenza** [reply]
Or maybe you already did it :)

[feature] add curl to images so that HEALTHCHECK is possible

Thank you!

**Reply by agoldis** [reply]
Sure, `ARG USER=node` seems like a good solution
